python/grass/jupyter/interactivemap.py

```diff
@@ -346,6 +346,190 @@ def add_layer_control(self, **kwargs):
+
+    #         save_button.on_click(save_geometries)
+
+    #         # Add control for the widgets
+    #         save_button_control = WidgetControl(
+    #             widget=widgets.VBox([name_input, save_button]), position="topright"
+    #         )
+    #         self.map.add_control(save_button_control)
+
+    #         display(out)
+
+    #     def hide_interface():
+    #         nonlocal save_button_control, name_input, save_button
+
+    #         out.clear_output()
+    #         if save_button_control:
+    #             self.map.remove_control(save_button_control)
+    #             save_button_control = None
+    #         # Remove the draw control if it's added
+    #         if draw_control in self.map.controls:
+    #             draw_control = GeomanDrawControl(
+    #                 drawOptions={
+    #                     'polygon': False,
+    #                     'rectangle': False,
+    #                     'circle': False,
+    #                     'marker': False,
+    #                     'polyline': False
+    #                 },
+    #                 editOptions={
+    #                     'edit': True,
+    #                     'remove': True
+    #                 },
+    #                 hide_controls=False  # Ensure the toolbar is visible
+    #             )
+    #             self.map.remove_control(draw_control)
+    #         # Remove the widget controls for the name input and save button
+    #         if name_input:
+    #             name_input.close()
+    #             name_input = None
+    #         if save_button:
+    #             save_button.close()
+    #             save_button = None
+
+    #     toggle_control = WidgetControl(
+    #         widget=draw_button, position="topright"
+    #     )
+    #     self.map.add_control(toggle_control)
+
+    def show_control(self):
+        import ipywidgets as widgets
+
+        # Create the toggle button
+        draw_button = widgets.ToggleButton(
+            icon="pencil",
+            value=False,
+            tooltip="Click to draw geometries",
+            layout=widgets.Layout(width="33px", margin="0px 0px 0px 0px"),
+        )
+
+        geoman_control = self._ipyleaflet.GeomanDrawControl(position="topright")
+
+        def on_toggle_change(change):
+            if change["new"]:
+                # Set drawOptions and editOptions when the button is toggled on
+                geoman_control.polyline = {"pathOptions": {}}
+                geoman_control.polygon = {"pathOptions": {}}
+                geoman_control.circlemarker = {"pathOptions": {}}
+                geoman_control.drag = False
+                geoman_control.cut = False
+                geoman_control.rotate = False
+                geoman_control.edit = False
+                geoman_control.remove = False
+                self.map.add_control(geoman_control)
+            else:
+                self.map.remove_control(geoman_control.polyline)
```

Comment:
Shouldn't you remove `geoman_control` instead of `geoman_control.polyline`? I haven't actually tested it, it's just a guess.

Reply:
BTW, `remove_control` is obsolete, use `remove`, see docs.

Reply:
This is probably some limitation on their side. Let's go back to DrawControl then.